### PR TITLE
Create iterator wrapped device loader for multi-processing.

### PR DESCRIPTION
**Pull request by dlibenzi**

### CODE REVIEW

**Comment by dlibenzi** [comment]
Fixed #2115

File: torch_xla/distributed/parallel_loader.py [hunk 1]
```diff
@@ -173,3 +173,26 @@ def _worker(self, dqueue):
...
+    loader (:class:`torch.utils.data.DataLoader`): The PyTorch DataLoader to be
+      wrapped.
+    device (`torch.device`...): The device where the data has to be sent.
+    kwargs: Named arguments for the `ParallelLoader` constructor.
+  """
+
+  def __init__(self, loader, device, **kwargs):
+    self._loader = loader
+    self._device = device
+    self._parallel_loader_args = kwargs
```
Comment (jysohn23) [inline torch_xla/distributed/parallel_loader.py:193]: Can we rename this as `_parallel_loader_kwargs`?

File: torch_xla/distributed/parallel_loader.py [hunk 2]
```diff
@@ -173,3 +173,26 @@ def _worker(self, dqueue):
...
+
+  def __iter__(self):
+    parallel_loader = ParallelLoader(self._loader, [self._device],
+                                     *self._parallel_loader_args)
```
Comment (jysohn23) [inline torch_xla/distributed/parallel_loader.py:197]: Shouldn't this be be `**self._parallel_loader_kwargs` (extra `*`)?

File: torch_xla/distributed/parallel_loader.py (hunk 1 above)
Comment (dlibenzi) [reply torch_xla/distributed/parallel_loader.py]: Yep

File: torch_xla/distributed/parallel_loader.py (hunk 2 above)
Comment (dlibenzi) [reply torch_xla/distributed/parallel_loader.py]: Duh!

**Comment by taylanbil** [comment]
Did you test this or does CI cover running these files?

**Comment by dlibenzi** [comment]
> Did you test this or does CI cover running these files?

Tested on Cloud TPU.

**Comment by vfdev-5** [comment]
@dlibenzi could you please add `__len__` method based on given `loader` ?

**Pull request merged**